src/repl-frontend/scala/tools/nsc/MainGenericRunner.scala

```diff
@@ -78,10 +78,51 @@ class MainGenericRunner {
           None
         case _  =>
           // We start the repl when no arguments are given.
-          // If user is agnostic about both -feature and -deprecation, turn them on.
+          // If user is agnostic about both -feature and -deprecation, turn them on and add batteries.
           if (settings.deprecation.isDefault && settings.feature.isDefault && settings.lint.isDefault) {
             settings.deprecation.value = true
             settings.feature.value = true
+            locally {
+              import settings.LintWarnings._
+              for (linted <- allLintWarnings if linted != Unused)
+                settings.lint.enable(linted)
+            }
+            locally {
+              import settings.UnusedWarnings._
+              val unused = List (
+                //Imports,  // not until REPL is fixed not to include spurious imports
```

Comment:
Even still.  An import that goes unused for a second shouldn't warn in a REPL, IMO.

And to be honest I'm not sure about adding and warning about all these things.  REPL code isn't production code.  If I define a pat var I don't want to be "yelled at" by the REPL about it.

IMO let's add the things we feel are just misleading not to warn about.  Or just add the imports, not the lints.

Reply:
In a REPL session, I am evolving some code. How to choose between "don't warn me, I'm just experimenting" and "please warn me so I evolve in the right direction"?

I'm leaning toward warning, so long as the presentation isn't verbose or interferes with normal session.

Reply:
Hmm, I see what you mean.  But maybe even when you're evolving some code, the REPL pointing out a bunch of lints is maybe too much.  Maybe removing the lints is a refinement process that you do after when you've move the code?

I'm not against adding enabling any lint warnings, but I don't think we have much room on change how verbose the presentation is.  I mean, unless we do something like make

```
scala> import scala.concurrent.Future
                               ^
       warning: Unused import
import scala.concurrent.Future
```

something like

```
scala> import scala.concurrent.Future
import scala.concurrent.Future (***)
```

Or do a warn once per session (with instructions for more info).

Reply:
> How to choose between "don't warn me, I'm just experimenting" and "please warn me so I evolve in the right direction"?

It's a tough call, but I think I'm with Dale. Let's see which lint / unused / goodie warnings are clear wins and have low risk for being noisy even when writing drafty code, which people certainly do a lot in the REPL.